esphome/components/tem3200/tem3200.cpp

```diff
@@ -0,0 +1,154 @@
+#include "tem3200.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace tem3200 {
+
+static const char *const TAG = "tem3200";
+
+enum ErrorCode {
+  NONE = 0,
+  RESERVED = 1,
+  STALE = 2,
+  FAULT = 3,
+};
+
+void TEM3200Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up TEM3200...");
+
+  uint8_t status(NONE);
+  uint16_t temperature_raw(0);
+  uint16_t pressure_raw(0);
+
+  // startup device delay
+  delay(10);
```

Comment:
is this actually to make sure the device has time to start before you talk to it?

Reply:
That is why I had put it there yes, I don't recall where I saw a 8ms time until the sensor is ready. I haven't seen it ever be a problem. Maybe I can wrap this up with the startup retry code.

Reply:
Given that other components don't seem to do this, I am going to leave this be for now.